lib/image_processing/processor.rb

```diff
@@ -20,7 +20,7 @@ def initialize(accumulator = nil)
     end
 
     def custom(&block)
-      (block && block.call(@accumulator)) || @accumulator
+      block_given? ? yield(@accumulator) : @accumulator
```

Comment:
The current version has an additional feature that the proposed one doesn't, which is that if the block returns `nil`, the `@accumulator` will be returned by the method. This is intentional as it allows the user not to have to return the accumulator object in the `custom` block.

So, I would prefer it to stay as it is. It could theoretically be written as

```rb
(block_given? && yield(@accumulator)) || accumulator
```

but I don't see the benefit.

Reply:
reverted